### PR TITLE
docs: provide docs for channel usage

### PR DESCRIPTION
**Pull request by Kalovelo**
For easier reviewing, check the formatted versions:

[README.md](https://github.com/aeternity/state-channel-demo/blob/df5efe70a6d209a291412632dccf52ca63cf3343/README.md)
[CHANNEL_USAGE.md](https://github.com/aeternity/state-channel-demo/blob/df5efe70a6d209a291412632dccf52ca63cf3343/CHANNEL_USAGE.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,3 +43,47 @@ cd client && npm run dev:testnet
...
+  - Generate a new account (bot player - channel `initiator`) for each request
+  - Fund both accounts by requesting coins from [faucet app](https://faucet.aepps.com/)
+  - Respond to users request with the mutual channel configuration. Read more about the channel configuration [here](https://github.com/aeternity/protocol/blob/master/node/api/channels_api_usage.md#channel-establishing-parameters)
+- Channel Initialization Phase
+  - When both accounts have enough funds, the server app will run `Channel.Initialize(channelConfig)` which will execute the on-chain transaction `channel_create_tx` and proceed in a listening-for-events state such as channel open confirmation
+  - Also, as the initiator, bots are responsible for deploying the contract on channel.
+- Game rounds 
+  - One channel configuration option is the `sign` function. This function runs whenever the other party executes an off-chain transaction which needs to be co-signed by both participants. Therefore whenever the other party makes their move, we can confirm it here and then respond with a next contract call.
```
Comment (martinkaintas) [inline README.md:61]:
```suggestion
  - One channel configuration option is the `sign` function. This function runs whenever any party executes an off-chain transaction which needs to be co-signed by both participants. Therefore whenever the other party makes their move, we can confirm it here and then respond with a next contract call.
```

File: CHANNEL_USAGE.md [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+```
+
+## Step 3: Request coins from faucet app
+### Initiator side (Similar to) Responder side
+```js
+axios.post(`https://faucet.aepps.com/account/${sdk.selectedAddress}`, {});
+```
+
+
+## Step 4: sign transactions
```
Comment (martinkaintas) [inline CHANNEL_USAGE.md:34]:
```suggestion
## Step 4: Sign transactions
```

File: CHANNEL_USAGE.md [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+These sign functions are part of the channel configuration. They are executed on each channel related transaction, and each one of them need to **be confirmed by both ends**.
```
Comment (martinkaintas) [inline CHANNEL_USAGE.md:35]:
```suggestion
These sign functions are part of the channel configuration. They are executed on each channel related transaction, and each one of them needs to **be confirmed by both ends**.
```

File: CHANNEL_USAGE.md [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+    registerEvents(channelInstance,savedState.channelConfig);
+  }
+```
+
+## Step 8: Closing channel
+There are 2 scenarions
+
+### Channel is mutually closed
+For example the responder would like to close the channel.
+In this case, the responder can simply execture `channel.shutdown`. The initiator will co-sign it and the channel will be closed.
```
Comment (martinkaintas) [inline CHANNEL_USAGE.md:208]:
```suggestion
In this case, the responder can simply execute `channel.shutdown`. The initiator will co-sign it and the channel will be closed.
```

**Review by martinkaintas — COMMENTED** [review]
Looks good to me, just a few typos 👀

**Comment by marc0olo** [comment]
@Kalovelo I will let @davidyuk do the review, I think he is the right person to look at this. especially as we plan to include that also in the SDK docs somehow as discussed in https://github.com/aeternity/aepp-sdk-js/issues/1244

**Comment by marc0olo** [comment]
@Kalovelo what's the state here? are we waiting for @davidyuk ?

**Comment by Kalovelo** [comment]
@marc0olo We've made refactorings past week, so I changed it to draft since we need to update the docs as well - expect it to be ready by Monday

**Review by martinkaintas — APPROVED** [review]
Changes look good to me! 🤝

**Comment by Kalovelo** [comment]
@davidyuk docs were updated, could you give them a view? thanks!

File: README.md [hunk 5]
```diff
@@ -44,3 +43,47 @@ cd client && npm run dev:testnet
...
+- Initialize a channel with the given configuration
+- Verify that the deployed contract has the correct source code
+- Respond to other party's actions
+- Execute `channel_close_mutual_tx`
+- Reconnect to channel in cases of window reload
+
+# Helpful Links
+- [On-Chain channel transactions](https://github.com/aeternity/protocol/blob/master/channels/ON-CHAIN.md)
+- [Off-Chain channel transactions](https://github.com/aeternity/protocol/blob/master/channels/OFF-CHAIN.md)
+- [Channels API usage](https://github.com/aeternity/protocol/blob/master/node/api/channels_api_usage.md#channel-timeout-values)
```
Comment (davidyuk) [inline README.md:84]:
```suggestion
- [Channels API usage](https://github.com/aeternity/protocol/blob/master/node/api/channels_api_usage.md)
```

**Review by davidyuk — COMMENTED** [review]
The first 3 steps look fine, later it is more a listing of important parts of code, not very descriptive.

Deposit and withdrawal look missed, as well as `createContract`.

**Comment by davidyuk** [comment]
Do we need a tutorial or a guide?

I think [Channels - intended usage](https://github.com/aeternity/protocol/blob/master/node/api/channels_api_usage.md) is already suitable as a guide, the only problem with it is that it is not aware of js sdk.

Let's make a tutorial, e.g. step-by-step walk-through to make a specific app. This would be easier to read to people with no experience in SC.

In any case, the SC topic is huge, proposing to split it into 3 parts:
- setup sc, deposit, update, withdraw, happy close
- contracts
- advanced cases: reconnection, dispute, forceProgress

So, the previous topic doesn't depend on the next one.

File: CHANNEL_USAGE.md [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+Utility functions used
+- [verifyContractBytecode](#verifyContractBytecode)
+- [buildContract](#buildContract)
+```js
+
+// helper variable to keep track of the last contract caller
+let lastContractCaller;
+
+async responderSignTx(
+    tag: string,
```
[the remark this answers is not in the document]
Comment (Kalovelo) [reply CHANNEL_USAGE.md:47]: Doesn't TypeScript provide more context (therefore more readability) though?

**Comment by Kalovelo** [comment]
> Let's make a tutorial, e.g. step-by-step walk-through to make a specific app. This would be easier to read to people with no experience in SC.


@davidyuk 

@martinkaintas workshop (https://github.com/aeternity/state-channel-workshop) on state channels is a step by step project which I believe does that

File: CHANNEL_USAGE.md [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+  })
+}
+```
+
+## `checkIfChannelIsStillOpen`
+In order to check  if a channel is still open, we can make a get request to  node.
+```js
+  async checkifChannelIsStillOpen(channelId) {
+    const response = await fetch(`${NODE_URL}/v3/channels/${channelId}`);
+    const result = await response.json();
```
[the remark this answers is not in the document]
Comment (Kalovelo) [reply CHANNEL_USAGE.md:373]: On the next line we check whether if the result has `id` field

**Comment by Kalovelo** [comment]
Merging as comments have been addressed along with the addition of `withdraw`, `deposit` and `createContract`

**Pull request merged**